File: TaskMasterAPI/models/checklist_item.py
from database import db


class ChecklistItem(db.Model):
    task_id = db.Column("taskId", db.Integer, db.ForeignKey('task.id'), primary_key=True)
    item_number = db.Column("itemNumber", db.Integer, primary_key=True)
    completed = db.Column(db.Boolean)
    description = db.Column(db.TEXT)

File: task_master/lib/domain/model/check_list_item.dart
class ChecklistItem {
  int taskId;
  int itemNumber;
  bool completed;
  String description;

  ChecklistItem({required this.taskId, required this.itemNumber, required this.completed, required this.description});

  factory ChecklistItem.fromJson(Map<String, dynamic> json) {
    return ChecklistItem(
      taskId: json['task_id'],
      itemNumber: json['item_number'],
      completed: json['completed'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'task_id': taskId,
      'item_number': itemNumber,
      'completed': completed,
      'description': description,
    };
  }
}
